packages/miniflare/src/plugins/hyperdrive/index.ts

```diff
@@ -71,13 +71,22 @@ export const HyperdriveSchema = z
 		return {
 			database: url.pathname.replace("/", ""),
 			user: url.username,
-			password: url.password,
+			password: decodeIfURLEncoded(url.password),
 			scheme: url.protocol.replace(":", ""),
 			host: url.hostname,
 			port: port,
 		};
 	});
 
+function decodeIfURLEncoded(str: string): string {
+	try {
+		const decoded = decodeURIComponent(str);
+		return str === decoded ? str : decoded;
```

Comment:
This line doesn't actually seem to be doing much, you could just always return the decoded value

Reply:
Thanks I was overthinking it.